### PR TITLE
Expose a download() method on the archive to download all contents in a version

### PR DESCRIPTION
**Pull request by e-e-e**

### CODE REVIEW

File: test/basic.js [hunk 1]
```diff
@@ -116,3 +116,43 @@ tape('provide keypair', function (t) {
     })
   })
 })
+
+tape.only('download a version', function (t) {
```
Comment (mafintosh) [inline test/basic.js]: can you remove the .only here?

File: index.js [hunk 2]
```diff
@@ -301,6 +301,49 @@ Hyperdrive.prototype.checkout = function (version) {
...
+Hyperdrive.prototype.download = function (cb) {
+  var downloadCount = 1
+  var self = this
+
+  download('/')
+
+  function download (entry) {
+    self.stat(entry, function (err, stat) {
+      if (err) {
+        self.emit('error', err)
```
Comment (mafintosh) [inline index.js]: Don't emit an error, just call the cb

File: index.js [hunk 3]
```diff
@@ -301,6 +301,49 @@ Hyperdrive.prototype.checkout = function (version) {
...
+      }
+      if (stat.isDirectory()) return downloadDir(entry, stat)
+      if (stat.isFile()) return downloadFile(entry, stat)
+    })
+  }
+
+  function downloadDir (dirname, stat) {
+    self.readdir(dirname, function (err, entries) {
+      if (err) {
+        self.emit('error', err)
```
Comment (mafintosh) [inline index.js]: Same as above

File: index.js (hunk 2 above)
Comment (e-e-e) [reply index.js]: ok, easy.

**Comment by mafintosh** [comment]
9.9.0

**Pull request merged**